src/components/SessionActivityTracker.js

```diff
@@ -41,18 +41,19 @@ class SessionActivityTracker extends React.Component {
     if (this.state.counter === null) { // counter is null if timeout value hasn't been fetched yet
       return
     }
-
-    this.setState(
-      state => ({
-        counter: state.counter - 1,
-        showTimeoutModal: state.counter <= TIME_TO_DISPLAY_MODAL,
-      }),
-      () => {
-        if (this.state.counter <= 0) {
-          this.props.onLogout()
+    if (!(this.props.config.get('userSessionTimeoutInterval') <= 0)) {
```

Comment:
I recommend you don't wrap the `this.setState` call.  Just have another:
```js
if (configInterval <= 0) {
  return
}
```
above the `this.setState` block.